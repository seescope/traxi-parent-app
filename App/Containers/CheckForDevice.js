import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { Image, Text, View } from 'react-native';
import * as Animatable from 'react-native-animatable';
import { Actions } from 'react-native-router-flux';

import Button from '../Components/Button';
import Background from '../Components/Background';
import HeaderText from '../Components/HeaderText';
import Spacing from '../Components/Spacing';
import { GREY } from '../Constants/Colours';
import { isSmallScreen } from '../Utils';

const style = {
  container: {
    paddingTop: 18,
    alignItems: 'center',
  },
  innerContainer: {
    paddingHorizontal: 16,
  },
  labelText: {
    fontFamily: 'Raleway-Regular',
    fontSize: isSmallScreen ? 12 : 14,
    color: GREY,
  },
  buttonsContainer: {
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
  },
  deviceImage: {
    height: 200,
    marginVertical: 8,
  },
};

const CheckForDevice = ({ kidName }) => (
  <Background>
    <Spacing height={16} />
    <View style={style.container}>
      <HeaderText>Get {kidName}'s device now</HeaderText>

      <Animatable.Image
        animation="bounceInLeft"
        resizeMode="contain"
        style={style.deviceImage}
        source={require('../Images/device_image.png')}
      />

      <View style={style.innerContainer}>
        <Text allowFontScaling={false} style={style.labelText}>
          To start monitoring
          {' '}
          {kidName}
          's Internet activity you're going to need to get their smartphone or tablet now.
          {' '}
        </Text>

        <Spacing height={16} />

        <Text allowFontScaling={false} style={style.labelText}>
          If you don't have it now, we can send you an email tomorrow to remind you to come back and start monitoring
          {' '}
          {kidName}
          's device.
        </Text>
      </View>

      <Spacing height={32} />

      <Animatable.View
        useNativeDriver
        style={style.buttonsContainer}
        animation="bounceIn"
        delay={500}
      >
        <Button onPress={() => Actions.sendReminder()}>
          Not ready yet
        </Button>
        <Button primary onPress={() => Actions.deviceSetup()}>
          I'm ready
        </Button>
      </Animatable.View>
    </View>
  </Background>
);

CheckForDevice.propTypes = {
  kidName: PropTypes.string.isRequired,
};

const mapStateToProps = state => {
  const { kidUUID } = state.setupState;
  const { name } = state.kidsState[kidUUID] || {};

  return {
    kidName: name || '',
  };
};

export default connect(mapStateToProps)(CheckForDevice);
