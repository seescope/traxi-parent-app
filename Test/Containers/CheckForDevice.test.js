import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import CheckForDevice from '../../App/Containers/CheckForDevice';

const mockStore = configureStore([thunk]);
const testStore = mockStore({
  kidsState: {
    'abc-123': {
      name: 'John Bobson',
    },
  },
  setupState: {
    kidUUID: 'abc-123',
  },
});

const CheckForDeviceComponent = () => (
  <Provider store={testStore}>
    <CheckForDevice />
  </Provider>
);

it('renders the <SetName> component', () => {
  const tree = renderer.create(<CheckForDeviceComponent />).toJSON();
  expect(tree).toMatchSnapshot();
});
