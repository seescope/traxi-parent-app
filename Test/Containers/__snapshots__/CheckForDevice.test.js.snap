// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the <SetName> component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F0F2F7",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": 16,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "paddingTop": 18,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#545454",
          "fontFamily": "Raleway-ExtraBold",
          "fontSize": 26,
          "textAlign": "center",
        }
      }
    >
      Get 
      John Bobson
      's device now
    </Text>
    <Image
      animation="bounceInLeft"
      resizeMode="contain"
      source={1}
      style={
        Object {
          "height": 250,
        }
      }
    />
    <View
      style={
        Object {
          "paddingHorizontal": 16,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#545454",
            "fontFamily": "Raleway-Regular",
            "fontSize": 14,
          }
        }
      >
        To start monitoring
         
        John Bobson
        's Internet activity you're going to need to get their smartphone or tablet now.
         
      </Text>
      <View
        style={
          Object {
            "height": 16,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#545454",
            "fontFamily": "Raleway-Regular",
            "fontSize": 14,
          }
        }
      >
        If you don't have it now, we can send you an email tomorrow to remind you to come back and start monitoring
         
        John Bobson
        's device.
      </Text>
    </View>
    <View
      style={
        Object {
          "height": 16,
        }
      }
    />
    <View
      animation="bounceIn"
      delay={500}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "flex-end",
        }
      }
      useNativeDriver={true}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 16,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#448AFF",
              "fontFamily": "Raleway-Regular",
              "fontSize": 20,
            }
          }
        >
          I'm ready
           
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 16,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#448AFF",
              "fontFamily": "Raleway-Regular",
              "fontSize": 14,
            }
          }
        >
          Not ready yet
           
        </Text>
      </View>
    </View>
  </View>
</View>
`;
